### ISSUE TITLE
when  pdf_renderer='hocr' mode the output PDF text  is reversed?

### ISSUE THREAD

**Issue by gogineniravikumar** [jbarlow83/OCRmyPDF#642]
**Describing the bug**
When I generate output PDF using using ocrmypdf in  pdf_renderer='hocr' mode the out put pdf  content stream  (text  is reversed  - in content stream the first line becomes last line and last line become you encounter first  in content stream ). Because of this,  when you tag the out put file it is reading reverse. 

**To Reproduce**
ocrmypdf.ocr("InputScanned.pdf", "outputwithtext.pdf", pdf_renderer='hocr', keep_temporary_files=True)

vs 
correct out put file you can generate using command
ocrmypdf.ocr("InputScanned.pdf", "outputwithtext.pdf" ,keep_temporary_files=True)

```bash
ocrmypdf.ocr("InputScanned.pdf", "outputwithtext.pdf",pdf_renderer='hocr', keep_temporary_files=True)
```

**Example file**
input [file ](https://drive.google.com/file/d/1bdgN-569x5YotQLlsGdDHy2VKG96aJex/view?usp=sharing)
error output [file ](https://drive.google.com/file/d/1IVewJo6V-12DQ6QVHjlpfrDeJncz4S-G/view?usp=sharing)

**Expected behavior**
After observation you can see that the text should be in correct order in text layer.  

**Screenshots**
output pdf content stream 
![image](https://user-images.githubusercontent.com/50137469/94533740-33d8ea80-025d-11eb-86cf-156e54b34662.png)

**System**
 - OS: Windows 10
 - OCRmyPDF Version: ``ocrmypdf --11.1.0``

**Comment by jbarlow83** [comment]
You showed me a screenshot of the order you expected from another OCR program, I think, with the concern that OCRmyPDF is presenting the content stream information in reverse order from what is shown there. 

Yes, we shouldn't do it exactly backwards and that is a good catch, and conveniently can be fixed by removing a minus sign.

Just to forewarn, you can't actually rely on the order in which text appears in a page content stream. The only reasonable way to interpret a content stream is parse it and fully locate the position of all text. On top of that, the characters in bytestrings in a content stream are glyph IDs that map a particular character in a particular font, so you can't even trust that the characters inside `(here ) Tj` will be legible. You have to use a program like poppler pdftotext or pdfminer.six that understands the intracies of PDF fonts. A PDF writer is free to write content in any order it pleases. In some cases, reordering the text can affect display, e.g. if text is written over top of other text. It sounds to me like you're just dealing with another program that is using the content stream for tagging order, and this may not matter, but it's a common issue people run into so I hand out reminders freely.

**Comment by gogineniravikumar** [comment]
I found why its happening. Its small fix. In hocrtransform.py file the 
`for line in sorted(
            chain(
                self.hocr.iterfind(self._child_xpath('span', 'ocr_header')),
                self.hocr.iterfind(self._child_xpath('span', 'ocr_line')),
                self.hocr.iterfind(self._child_xpath('span', 'ocr_textfloat')),
            ),
            key=self.topdown_position,
        ):`
You have to reverse the sorted function out put
`for line in sorted(
            chain(
                self.hocr.iterfind(self._child_xpath('span', 'ocr_header')),
                self.hocr.iterfind(self._child_xpath('span', 'ocr_line')),
                self.hocr.iterfind(self._child_xpath('span', 'ocr_textfloat')),
            ),
            key=self.topdown_position,
        ).reverse():`

Then the pdf's content stream in sorted order. Please update this in next release.

**Comment by jbarlow83** [comment]
@gogineniravikumar This was fixed in v11.1.2.
